Marlin/UBL_G29.cpp

```diff
@@ -712,58 +712,193 @@
           z_values[x][y] += constant;
   }
 
+  bool position_is_reachable_UBL(float X_target, float Y_target , bool is_probe) {
+
+    float dx = RAW_X_POSITION(X_target),
+          dy = RAW_Y_POSITION(Y_target);
+
+    #if IS_SCARA
+      #if MIDDLE_DEAD_ZONE_R > 0
+        const float R2 = HYPOT2(dx - SCARA_OFFSET_X, dy - SCARA_OFFSET_Y);
+        return R2 >= sq(float(MIDDLE_DEAD_ZONE_R)) && R2 <= sq(L1 + L2);
+      #else
+        return HYPOT2(dx - SCARA_OFFSET_X, dy - SCARA_OFFSET_Y) <= sq(L1 + L2);
+      #endif
+    #elif ENABLED(DELTA)
+      #if defined(DELTA_RADIUS)
+        #if !defined(DELTA_TRAVEL_RADIUS)
+          #define DELTA_TRAVEL_RADIUS DELTA_RADIUS
+        #endif
+        float radius = DELTA_RADIUS - UBL_MESH_INSET;
+        if (is_probe) {
+          dx -= X_PROBE_OFFSET_FROM_EXTRUDER;
+          dy -= Y_PROBE_OFFSET_FROM_EXTRUDER;
+          radius = DELTA_TRAVEL_RADIUS;
+        }
+        return HYPOT2(dx, dy) <= sq((float)(radius) + 0.001);
+      #else
+        SERIAL_PROTOCOLLNPG("ERROR - DELTA_RADIUS not defined");
+        return false;
+      #endif
+
+    #else
+      float x_min_pos = UBL_MESH_MIN_X;
+      float x_max_pos = UBL_MESH_MAX_X;
+      float y_min_pos = UBL_MESH_MIN_Y;
+      float y_max_pos = UBL_MESH_MAX_Y;
+      if (is_probe) {
+        dx -= X_PROBE_OFFSET_FROM_EXTRUDER;
+        dy -= Y_PROBE_OFFSET_FROM_EXTRUDER;

```

Comment:
This doesn't work for SCARA, because the probe offset changes as the arm rotates.